src/main/java/org/broadinstitute/consent/http/resources/TDRResource.java

```diff
@@ -64,4 +86,64 @@ public Response getDatasetByIdentifier(@Auth AuthUser authUser, @PathParam("iden
             return createExceptionResponse(e);
         }
     }
+
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("/dar/draft")
+    @PermitAll
+    public Response createDraftDataAccessRequest(
+        @Auth AuthUser authUser,
+        @Context UriInfo info,
+        @QueryParam("identifiers") String identifiers,
+        @QueryParam("projectTitle") String projectTitle) {
+      try {
+        // Ensure that the user is a registered DUOS and SAM user
+        User user = userService.findOrCreateUser(authUser);
+        if (Objects.isNull(identifiers) || identifiers.isBlank()) {
+          throw new BadRequestException("No dataset identifiers were provided");
+        } else {
+          List<Integer> identifierList = Arrays.stream(identifiers.split(","))
+                  .map(String::trim)
+                  .filter(identifier -> !identifier.isBlank())
+                  .map(Dataset::parseIdentifierToAlias)
+                  .toList();
+          List<Dataset> datasets = tdrService.getDatasetsByIdentifier(identifierList);
+          List<Integer> datasetAliases = datasets
+                  .stream()
+                  .map(Dataset::getAlias)
+                  .toList();
+        // Check that we were able to find a dataset id for all identifiers provided
+        if (identifierList.size() != datasets.size()) {
+            // isolate a list of identifiers that were NOT found
+            List<Integer> notFoundIdentifiers = identifierList
+                    .stream()
+                    .filter(identifier -> !datasetAliases.contains(identifier))
+                    .toList();
+            // throw a NFE to let the client know which identifiers were NOT found so they can rectify their request
+            throw new NotFoundException("Invalid dataset identifiers were provided: " + notFoundIdentifiers);
+        }
+        List<Integer> datasetIds = datasets
+                .stream()
+                .map(Dataset::getDataSetId)
+                .toList();
+        DataAccessRequest newDar = new DataAccessRequest();
+        DataAccessRequestData data = new DataAccessRequestData();
+        String referenceId = UUID.randomUUID().toString();
+        newDar.setReferenceId(referenceId);
+        data.setReferenceId(referenceId);
+        if (!Objects.isNull(projectTitle) && !projectTitle.isBlank()) {
+          data.setProjectTitle(projectTitle);
+        }
+        newDar.setData(data);
+        newDar.setDatasetIds(datasetIds);
+        DataAccessRequest result = darService.insertDraftDataAccessRequest(user, newDar);
+        // URI should return the new DAR url
+        URI uri = info.getRequestUriBuilder().replacePath("").replacePath("api/dar/v2/" + result.getReferenceId()).replaceQuery(null).build();
```

Comment:
I was looking into this some more, I think an easier way to do this is to use the `UriBuilder.fromPath()` method instead:
```suggestion
        URI uri = UriBuilder.fromPath("api/dar/v2/" + result.getReferenceId()).build();
```
This also means you don't need the `@Context UriInfo info` in the method signature.

Reply:
This looks like it's still an issue after the latest commit.

Reply:
Updating the URI to the suggested logic returns the correct sub path but not the main part of the URL (at least according to my unit tests when debugging).
With current changes, URI is:
`https://test.domain.org/api/dar/v2/<reference_id>`
With the suggested changes, URI would be:
`api/dar/v2/<reference_id>`
Is that intended?

Reply:
Refactored the URI with the suggested change.